pkg/cmd/roachtest/operations/node_kill.go

```diff
@@ -0,0 +1,122 @@
+	db, err := c.ConnE(ctx, o.L(), cl.nodes[0])
+	if err != nil {
+		c.Run(ctx, option.WithNodes(cl.nodes), "./cockroach.sh")
+		return
+	}
+	defer db.Close()
+	_, err = db.Query("SELECT 1")
+	if err != nil {
+		c.Run(ctx, option.WithNodes(cl.nodes), "./cockroach.sh")
+	}
+}
+
+func nodeKillRunner(signal int, drain bool) func(ctx context.Context, o operation.Operation, c cluster.Cluster) registry.OperationCleanup {
+	return func(ctx context.Context, o operation.Operation, c cluster.Cluster) registry.OperationCleanup {
+		return runNodeKill(ctx, o, c, signal, drain)
+	}
+}
+
+func runNodeKill(
+	ctx context.Context, o operation.Operation, c cluster.Cluster, signal int, drain bool,
+) registry.OperationCleanup {
+	rng, _ := randutil.NewPseudoRand()
+
+	nodes := c.All()
+	nid := nodes[rng.Intn(len(nodes))]
+
+	if drain {
+		o.Status(fmt.Sprintf("draining node %d", nid))
+
+		url, err := c.ExternalPGUrl(ctx, o.L(), c.Node(nid), roachprod.PGURLOptions{
+			External: false,
+			Secure:   c.IsSecure(),
+		})
+		if err != nil {
+			o.Fatal(err)
+		}
+		err = c.RunE(ctx, option.WithNodes(c.Node(nid)),
+			o.ClusterCockroach(), "node", "drain", "--logtostderr=INFO",
+			fmt.Sprintf("--url=%s", url[0]), "--self",
+		)
+		if err != nil {
+			o.Fatal(err)
+		}
+	}
+	o.Status(fmt.Sprintf("killing node %d with signal %d", nid, signal))
+
+	stopOpts := option.DefaultStopOpts()
+	stopOpts.RoachprodOpts.Sig = signal
+	if err := c.StopE(ctx, o.L(), stopOpts, c.Node(nid)); err != nil {
+		o.Fatal(err)
+	}
+
+	return &cleanupNodeKill{
+		nodes: c.Node(nid),
+	}
+}
+
+func registerNodeKill(r registry.Registry) {
+	r.AddOperation(registry.OperationSpec{
+		Name:             "node-kill/sigkill/drain=true",
+		Owner:            registry.OwnerServer,
+		Timeout:          10 * time.Minute,
+		CompatibleClouds: registry.AllClouds,
+		Dependencies:     []registry.OperationDependency{registry.OperationRequiresZeroUnderreplicatedRanges},
+		Run:              nodeKillRunner(9 /* signal */, true /* drain */),
+	})
+	r.AddOperation(registry.OperationSpec{
+		Name:             "node-kill/sigkill/drain=false",
+		Owner:            registry.OwnerServer,
+		Timeout:          10 * time.Minute,
+		CompatibleClouds: registry.AllClouds,
+		Dependencies:     []registry.OperationDependency{registry.OperationRequiresZeroUnderreplicatedRanges},
+		Run:              nodeKillRunner(9 /* signal */, false /* drain */),
+	})
+	r.AddOperation(registry.OperationSpec{
+		Name:             "node-kill/sigterm/drain=true",
+		Owner:            registry.OwnerServer,
+		Timeout:          10 * time.Minute,
+		CompatibleClouds: registry.AllClouds,
+		Dependencies:     []registry.OperationDependency{registry.OperationRequiresZeroUnderreplicatedRanges},
+		Run:              nodeKillRunner(15 /* signal */, true /* drain */),
+	})
+	r.AddOperation(registry.OperationSpec{
+		Name:             "node-kill/sigterm/drain=false",
+		Owner:            registry.OwnerServer,
```

Comment:
Not sure we should be assigning anything to the server "team"? 😅

Reply:
Haha, good point. I copied this from one of the roachtests. Any idea on what other team can own this? (We don't auto-create issues for operations so this means little other than for our own tracking).

Reply:
> I copied this from one of the roachtests

Which one? A basic search doesn't reveal any roachtests owned by server, but maybe ownership is assigned in some indirect way.

> Any idea on what other team can own this?

Maybe server is fine. I guess eventually these issue will be owned by a "DRT" team that will have to investigate and triage appropriately.

Reply:
> Which one? A basic search doesn't reveal any roachtests owned by server, but maybe ownership is assigned in some indirect way.

Oops, yeah I wasn't searching right. And yeah we can keep it assigned to server for now, with the understanding that all operation failures get triaged by DRT anyway.